Build fails for Ubuntu 17.10

Here it is:

```
[100%] Linking CXX executable ../../bin/RoR
/usr/bin/ld: warning: libboost_system.so.1.58.0, needed by /usr/lib/gcc/x86_64-linux-gnu/7/../../../../lib/libCaelum.so, not found (try using -rpath or -rpath-link)
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/7/../../../../lib/libPagedGeometry.a(BatchPage.cpp.o): relocation R_X86_64_32 against hidden symbol `_ZTVN7Forests9BatchPageE' can not be used when making a shared object
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/7/../../../../lib/libPagedGeometry.a(BatchedGeometry.cpp.o): relocation R_X86_64_32 against hidden symbol `_ZTVN7Forests15BatchedGeometryE' can not be used when making a shared object
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/7/../../../../lib/libPagedGeometry.a(GrassLoader.cpp.o): relocation R_X86_64_32 against hidden symbol `_ZTVN7Forests11GrassLoaderE' can not be used when making a shared object
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/7/../../../../lib/libPagedGeometry.a(ImpostorPage.cpp.o): relocation R_X86_64_32 against hidden symbol `_ZTVN7Forests12ImpostorPageE' can not be used when making a shared object
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/7/../../../../lib/libPagedGeometry.a(PagedGeometry.cpp.o): relocation R_X86_64_32 against symbol `_ZN4Ogre7Vector34ZEROE' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/7/../../../../lib/libPagedGeometry.a(PropertyMaps.cpp.o): relocation R_X86_64_32 against symbol `_ZN4Ogre20ResourceGroupManager27DEFAULT_RESOURCE_GROUP_NAMEB5cxx11E' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/7/../../../../lib/libPagedGeometry.a(StaticBillboardSet.cpp.o): relocation R_X86_64_32 against `.rodata' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/7/../../../../lib/libPagedGeometry.a(TreeLoader2D.cpp.o): relocation R_X86_64_32 against hidden symbol `_ZTVN7Forests12TreeLoader2DE' can not be used when making a shared object
/usr/bin/ld: final link failed: Nonrepresentable section on output
collect2: error: ld returned 1 exit status
source/main/CMakeFiles/RoR.dir/build.make:4926: recipe for target 'bin/RoR' failed
make[2]: *** [bin/RoR] Error 1
CMakeFiles/Makefile2:201: recipe for target 'source/main/CMakeFiles/RoR.dir/all' failed
make[1]: *** [source/main/CMakeFiles/RoR.dir/all] Error 2
Makefile:151: recipe for target 'all' failed
make: *** [all] Error 2
```

FYI here are libboost-system version available:

```
libboost-system1.62-dev
libboost-system1.63-dev
```

Maybe switching back to compiling dependencies might be more portable.

--- a/README.md
+++ b/README.md
@@ -1,34 +1,35 @@
 # Rigs of Rods build scripts
 These shell scripts build the vehicle simulator Rigs of Rods and its dependencies on systems using *apt* as package manager (e.g. Debian, Ubuntu, Mint).
-With some minor adjustments they should work on Linux distributions not based on *apt* as well.
+With some minor adjustments, they should work on Linux distributions not based on *apt* as well.
 
 ### Build and install Rigs of Rods:
-Run these commands from console in this order and not simultaneously:
+Run these commands from the console in this order and not simultaneously:
+
+`sh dependencies-Ubuntu-16.04.sh` if your distro is based on Ubuntu 16.04 OR `dependencies.sh` if your distro is based on something else.
 ```sh
-sh dependencies.sh
-sh core.sh
+sh game.sh
 sh content.sh
 # if you want to you can also install the community-map, but it's still WIP
 sh community_map.sh
 ```
 
 ##### Notes:
-* `dependencies.sh` will first ask for sudo password and will prompt you a bit later to confirm the installation of the packages. Make sure the packages don't conflict with something you need (usually doesn't happen).
-* `core.sh` don't require user interaction.  
+* `dependencies.sh` and `dependencies-Ubuntu-16.04.sh` will first ask for sudo password and will prompt you a bit later to confirm the installation of the packages. Make sure the packages don't conflict with something you need (usually doesn't happen).
+* `game.sh` asks for sudo password to install the game.  
 * `content.sh` only requires you to choose the content you want to download in the beginning.
 * To change default directories and *make* parameters edit the `config` file (optional).
-* Depending on your hardware and internet speed this can take quite a while. As long as the scripts are running everything is fine. If an errors occurs they will stop and state the error.
+* Depending on your hardware and internet speed this can take quite a while. As long as the scripts are running everything is fine. If an error occurs they will stop and state the error.
 
 
 ### Update to current snapshot
 simply re-run
 ```sh
-sh core.sh
+sh game.sh
 ```
 
 ##### General notes:
 * Included is a script for profiling. If you don't know what that means then it is not for you!
-* Scripts to build the RoR multiplayer server and set up a development environment for the RoR website named `server.sh` and `website.sh`  respecitvely are included as well (see the scripts for additional information).
+* Scripts to build the RoR multiplayer server and set up a development environment for the RoR website named `server.sh` and `website.sh`  respectively are included as well (see the scripts for additional information).
 * If for whatever reason you copy parts of the script into the console make sure to set configuration
 variables appropriately, either manually, or by *sourcing* the `config` file like this:
 ```

--- a/config
+++ b/config
@@ -1,14 +1,17 @@
 # The directory source files are downloaded and extracted to.
-ROR_SOURCE_DIR=~/rigs-of-rods/source
+ROR_SOURCE_DIR=~/rigs-of-rods-source
 
-# The directory binaries are installed to.
-ROR_INSTALL_DIR=~/rigs-of-rods/install
+# selects CMAKE_BUILD_TYPE, compiled objects will be located in $ROR_SOURCE_DIR/0_build/CMAKEBUILDTYPE
+CMAKEBUILDTYPE=RelWithDebInfo
+
+# change this if you want to hack on the code and have the repository forked to your GitHub account
+YOUR_GITHUB_USERNAME="RigsOfRods"
 
 # Make options. Defaults are:
 # -s       -- Silent operation; do not print the commands as they are executed.
-# -jX      -- Number of jobs to run simultaneously. X is the number of processors in /proc/cpuinfo by default.
-ROR_MAKEOPTS="-s -j$(nproc)"
+# -jX      -- Number of jobs to run simultaneously. X is the number of processors in /proc/cpuinfo by default(-j$(nproc)).
+ROR_MAKEOPTS="-s -j4"
 
 # Don't change these unless you know what you are doing
-PKG_CONFIG_PATH="$ROR_INSTALL_DIR/lib/pkgconfig"
-export PKG_CONFIG_PATH
+# PKG_CONFIG_PATH="$ROR_SOURCE_DIR/lib/pkgconfig"
+# export PKG_CONFIG_PATH

--- a/dependencies-Ubuntu-16.04.sh
+++ b/dependencies-Ubuntu-16.04.sh
@@ -1,0 +1,27 @@
+#!/bin/sh
+set -eu
+
+# Precompiled dependencies
+sudo apt-get update
+sudo apt-get -q install build-essential wget git wget cmake pkg-config python3 \
+libogre-1.9-dev libmygui-dev libmygui.ogreplatform0debian1v5 libopenal-dev \
+libcurl4-openssl-dev libgtk2.0-dev libois-dev libssl-dev libwxgtk3.0-dev \
+libssl-dev libcurl4-openssl-dev libgtk2.0-dev libwxgtk3.0-dev
+
+# Dependencies not found on official repos
+cd /tmp/
+sudo wget -nv http://ftp.debian.org/debian/pool/main/r/rapidjson/rapidjson-dev_1.1.0+dfsg-3_all.deb \
+http://prdownloads.sourceforge.net/rigs-of-rods/pagedgeometry-1.2.0-x86_64.deb \
+http://prdownloads.sourceforge.net/rigs-of-rods/socketw-3.10.26-x86_64.deb \
+http://prdownloads.sourceforge.net/rigs-of-rods/angelscript-2.31.2-x86_64.deb \
+http://prdownloads.sourceforge.net/rigs-of-rods/caelum-0.6.3-x86_64.deb
+
+sudo dpkg -i rapidjson-dev_1.1.0+dfsg-3_all.deb \
+pagedgeometry-1.2.0-x86_64.deb \
+socketw-3.10.26-x86_64.deb \
+angelscript-2.31.2-x86_64.deb \
+caelum-0.6.3-x86_64.deb
+
+
+echo "$(tput setaf 2)All dependencies were installed successfully"
+echo "You can now proceed with game.sh$(tput sgr 0)"

--- a/dependencies.sh
+++ b/dependencies.sh
@@ -1,27 +1,29 @@
-#!/bin/sh
+﻿#!/bin/sh
 set -eu
+. ./config
 
 # Precompiled dependencies
 sudo apt-get update
-sudo apt-get -q install build-essential wget git wget cmake pkg-config python3 \
-libogre-1.9-dev libmygui-dev libmygui.ogreplatform0debian1v5 libopenal-dev \
-libcurl4-openssl-dev libgtk2.0-dev libois-dev libssl-dev libwxgtk3.0-dev \
-libssl-dev libcurl4-openssl-dev libgtk2.0-dev libwxgtk3.0-dev
+sudo apt-get -q install build-essential git cmake pkg-config libboost-all-dev mercurial \
+libfreetype6-dev libfreeimage-dev libzzip-dev libois-dev \
+libgl1-mesa-dev libglu1-mesa-dev nvidia-cg-toolkit libopenal-dev  \
+libx11-dev libxt-dev libxaw7-dev libxrandr-dev \
+libssl-dev libcurl4-openssl-dev libgtk2.0-dev libwxgtk3.0-dev \
+libasound2-dev libpulse-dev wget
 
-# Dependencies not found on official repos
-cd /tmp/
-sudo wget -nv http://ftp.debian.org/debian/pool/main/r/rapidjson/rapidjson-dev_1.1.0+dfsg-3_all.deb \
-http://prdownloads.sourceforge.net/rigs-of-rods/pagedgeometry-1.2.0-x86_64.deb \
-http://prdownloads.sourceforge.net/rigs-of-rods/socketw-3.10.26-x86_64.deb \
-http://prdownloads.sourceforge.net/rigs-of-rods/angelscript-2.31.2-x86_64.deb \
-http://prdownloads.sourceforge.net/rigs-of-rods/caelum-0.6.3-x86_64.deb
+# Initialization
+if [ ! -e "$ROR_SOURCE_DIR" ]; then
+  mkdir -p "$ROR_SOURCE_DIR"
+fi
 
-sudo dpkg -i rapidjson-dev_1.1.0+dfsg-3_all.deb \
-pagedgeometry-1.2.0-x86_64.deb \
-socketw-3.10.26-x86_64.deb \
-angelscript-2.31.2-x86_64.deb \
-caelum-0.6.3-x86_64.deb
+cd "$ROR_SOURCE_DIR"
+if [ ! -e ror-dependencies ]; then
+  git clone https://github.com/${YOUR_GITHUB_USERNAME}/ror-dependencies.git
+fi
+cd ror-dependencies
 
+cmake .
+make $ROR_MAKEOPTS
 
-echo "$(tput setaf 2)All dependencies were installed successfully"
-echo "You can now proceed with core.sh$(tput sgr 0)"
+echo "$(tput setaf 2)All dependencies were installed and build successfully."
+echo "You can now proceed with game.sh$(tput sgr 0)"

--- a/game.sh
+++ b/game.sh
@@ -2,16 +2,14 @@
 set -eu
 . ./config
 
-#selects CMAKE_BUILD_TYPE, compiled objects will be located in $ROR_SOURCE_DIR/0_build/CMAKEBUILDTYPE
-CMAKEBUILDTYPE=RelWithDebInfo
-
-# change this if you want to hack on the code and have the repository forked to your GitHub account
-YOUR_GITHUB_USERNAME="RigsOfRods"
-
+# Initialization
+if [ ! -e "$ROR_SOURCE_DIR" ]; then
+  mkdir -p "$ROR_SOURCE_DIR"
+fi
 
 cd "$ROR_SOURCE_DIR"
 if [ ! -e rigs-of-rods ]; then
-  git clone https://github.com/${YOUR_GITHUB_USERNAME}/rigs-of-rods.git
+  git clone https://github.com/${YOUR_GITHUB_USERNAME}/rigs-of-rods.git --depth=5
 fi
 cd rigs-of-rods
 git pull
@@ -21,11 +19,17 @@ if [ ! -e "0_build/$CMAKEBUILDTYPE" ]; then
 fi
 cd 0_build/$CMAKEBUILDTYPE
 
+ROR_CMAKE_ARGS=""
+if [ -e "$ROR_SOURCE_DIR/ror-dependencies" ]; then
+    ROR_CMAKE_ARGS=-DCMAKE_PREFIX_PATH=$ROR_SOURCE_DIR/ror-dependencies/Dependencies_Linux/
+fi
+
 cmake ../../ \
--DCMAKE_INSTALL_PREFIX="$ROR_INSTALL_DIR" \
--DROR_BUILD_CONFIGURATOR:BOOL="TRUE" \
 -DCMAKE_BUILD_TYPE=$CMAKEBUILDTYPE \
--DCMAKE_CXX_FLAGS="-pipe -march=native"
+-DBUILD_CUSTOM_VERSION=ON \
+-DCUSTOM_VERSION="0.4.8.0-shscript" \
+-DCMAKE_CXX_FLAGS="-pipe -march=native" \
+$ROR_CMAKE_ARGS
 
 # CMAKE_CXX_FLAGS (flags for compiler) - Default are:
 # -march=native  -- Optimize RoR for the CPU the build is performed on; executable will only be portable to systems with same or newer CPU architecture
@@ -36,14 +40,9 @@ cmake ../../ \
 
 
 make $ROR_MAKEOPTS
-cp -R ../../bin .
-sed -i '/^PluginFolder=/d' bin/plugins.cfg
-echo "PluginFolder=$ROR_INSTALL_DIR/lib/OGRE" >>bin/plugins.cfg
-
-# there's no make install target, so just copy the bin folder
-cp -R bin "$ROR_INSTALL_DIR"
+sudo make install
 
 echo "$(tput setaf 2)Build succeeded!"
 echo "NOTE: Do not forget to run RoRConfig once before RoR."
-echo "NOTE: Binaries to start the game are in $ROR_INSTALL_DIR/bin"
-echo "NOTE: Mods belong in ~/.rigsofrods/packs$(tput sgr 0)"
+echo "NOTE: Mods belong in ~/.rigsofrods/packs"
+echo "NOTE: Run content.sh if you want to install the Content Pack$(tput sgr 0)"

